resource/abstractions/auditing_rules.md

```diff
@@ -0,0 +1,478 @@
+
+#### 📤 Returns
+* `Union[dict, bytes]`<br>
+  Returns a **Security Result Steps dictionary** or **Concatenated Security Request XML bytes** if the `ResourceAdmin.generate_requests_only` class attribute is set to `True`.
+
+#### ❌ Raises
+* `SecurityRequestError`<br>
+  Raises `SecurityRequestError` when the **SAF Return Code** of a **Security Result** is equal to `4`.
+* `DownstreamFatalError`<br>
+  Raises `DownstreamFatalError` when the **SAF Return Code** of a **Security Result** is greater than `4`.
+* `ValueError`<br>
+  Raises `ValueError` when values passed for `success`, `failure` or `all` are do not represent a valid `access level`.
+* `AlterOperationError`<br>
+  Raises `AlterOperationError` when the **general resource profile** cannot be altered because it does **NOT** exist.
+
+#### 💻 Example
+
+###### Python REPL
+```python
+>>> from pyracf import ResourceAdmin
+>>> resource_admin = ResourceAdmin()
+>>> resource_admin.overwrite_audit_by_attempt("TESTING","ELIJTEST",success="alter")
+{"step1":{"securityResult":{"resource":{"name":"TESTING","class":"ELIJTEST","operation":"set","requestId":"ResourceRequest","info":["Definition exists. Add command skipped due  to precheck option"],"commands":[{"safReturnCode":0,"returnCode":0,"reasonCode":0,"image":"RALTER  ELIJTEST             (TESTING)  AUDIT( SUCCESS (alter       ))","messages":["ICH11009I RACLISTED PROFILES FOR ELIJTEST WILL NOT REFLECT THE UPDATE(S) UNTIL A SETROPTS REFRESH IS ISSUED."]}]},"returnCode":0,"reasonCode":0}}}
+```
+
+###### Security Result Steps Dictionary as JSON
+```json
+{
+  "step1": {
+    "securityResult": {
+      "resource": {
+        "name": "TESTING",
+        "class": "ELIJTEST",
+        "operation": "set",
+        "requestId": "ResourceRequest",
+        "info": [
+          "Definition exists. Add command skipped due  to precheck option"
+        ],
+        "commands": [
+          {
+            "safReturnCode": 0,
+            "returnCode": 0,
+            "reasonCode": 0,
+            "image": "RALTER  ELIJTEST             (TESTING)  AUDIT( SUCCESS (alter       ))",
+            "messages": [
+              "ICH11009I RACLISTED PROFILES FOR ELIJTEST WILL NOT REFLECT THE UPDATE(S) UNTIL A SETROPTS REFRESH IS ISSUED."
+            ]
+          }
+        ]
+      },
+      "returnCode": 0,
+      "reasonCode": 0
+    }
+  }
+}
+```
+
+## `ResourceAdmin.overwrite_audit_by_access_level()`
+
+```python
+def overwrite_audit_by_access_level(
+    self,
+    resource: str,
+    class_name: str,
+    alter: Union[str, None] = None,
+    control: Union[str, None] = None,
+    read: Union[str, None] = None,
+    update: Union[str, None] = None,
+) -> Union[dict, bytes]:
+```
+
+#### 📄 Description
+
+Clear any rules defined within the **Audit Rules** for a general resource profile and replace them with a rule auditing `alter`, `control`, `read`, or `update` access attempts.
```

Comment:
Remove all currently defined **Auditing Rules** for a general resource profile and replace them with the new **Auditing Rules** specified by **Access Level**.

Reply:
covered by earlier changes.